src/libraries/Common/tests/System/Security/Cryptography/X509Certificates/CertificateAuthority.cs

```diff
@@ -823,14 +849,19 @@ internal static void BuildPrivatePki(
             // default to client
             extensions ??= new X509ExtensionCollection() { s_eeConstraints, s_eeKeyUsage, s_tlsClientEku };
 
-            using (RSA rootKey = RSA.Create(keySize))
-            using (RSA eeKey = RSA.Create(keySize))
+            if (keyFactory is null)
             {
-                var rootReq = new CertificateRequest(
-                    BuildSubject("A Revocation Test Root", testName, pkiOptions, pkiOptionsInSubject),
-                    rootKey,
-                    HashAlgorithmName.SHA256,
-                    RSASignaturePadding.Pkcs1);
+                // Should we add variance here?  Sometimes default to ML-DSA, sometimes EC-DSA, sometimes RSA?
```

Comment:
I'm of two minds here. On one hand, I like the idea of greasing things to keep things well exercised. On the other hand, I don't love nondeterministic failure. Can we hash a couple of the inputs like the `subject`, `testName`, and look at a couple of the bits to pick an algorithm? That way each test will, hopefully, keep using the same algorithm so they use the algorithm deterministically.

Reply:
Yeah, that's the same debate I was having internally.

Deterministic hashing of pkiOptions/intermediateAuthorityCount/testName/subjectName as an index into viable options seems reasonable to me.